src/modules/keyboardmanager/common/KeyboardManagerState.cpp

```diff
@@ -115,81 +115,120 @@ bool KeyboardManagerState::AddSingleKeyRemap(const DWORD& originalKey, const DWO
 }
 
 // Function to set the textblock of the detect shortcut UI so that it can be accessed by the hook
-void KeyboardManagerState::ConfigureDetectShortcutUI(const TextBlock& textBlock)
+void KeyboardManagerState::ConfigureDetectShortcutUI(const StackPanel& textBlock)
 {
-    std::lock_guard<std::mutex> lock(currentShortcutTextBlock_mutex);
-    currentShortcutTextBlock = textBlock;
+    std::lock_guard<std::mutex> lock(currentShortcutUI_mutex);
+    currentShortcutUI = textBlock;
 }
 
 // Function to set the textblock of the detect remap key UI so that it can be accessed by the hook
-void KeyboardManagerState::ConfigureDetectSingleKeyRemapUI(const TextBlock& textBlock)
+void KeyboardManagerState::ConfigureDetectSingleKeyRemapUI(const StackPanel& textBlock)
 {
-    std::lock_guard<std::mutex> lock(currentSingleKeyRemapTextBlock_mutex);
-    currentSingleKeyRemapTextBlock = textBlock;
+    std::lock_guard<std::mutex> lock(currentSingleKeyUI_mutex);
+    currentSingleKeyUI = textBlock;
+}
+
+
+void KeyboardManagerState::AddKeyToLayout(const StackPanel& panel, const hstring& key)
+{
+    // Textblock to display the detected key
+    TextBlock remapKey;
+    Border border;
+
+    border.Padding({ 20, 10, 20, 10 });
+    border.Margin({0, 0, 10, 0 });
+    border.Background(Windows::UI::Xaml::Media::SolidColorBrush{ Windows::UI::Colors::LightGray() });

```

Comment:
Could we use the AccentColor brush as the background of a key?

This way it would be inline with the settings UI where the accent color is used as the background and white as foreground.